src/main/java/models/Deadline.java

```diff
@@ -0,0 +1,32 @@
+package models;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
```

Comment:
You could use more detailed variable names, that could help a reader immediately see what your variable is supposed to represent. Don't be afraid to use slightly lengthier names for clarity! I noticed that some of your variable names are rather brief.